### ISSUE TITLE
Redirect to /connect/authorize IdP using User Switching Plugin + SSO enabled

### ISSUE THREAD

**Issue by alextochetto** [oidc-wp/openid-connect-generic#564]
**Describe the bug**
After login in a IdP, use the User Switching plugin to impersonate another user and a happens to /connect/authorize of IdP. This redirect authenticate the same user and the impersonate comes back to the logged user. If the login uses the button in form login, this scenario doesn't happen and everything works fine.

**To Reproduce**
1. Install User Switching plugin (https://wordpress.org/plugins/user-switching)
2. Configure the OpenID Connect Generic 
3. Enable SSO
4. Authenticate with user 'X'
5. Switch to user 'Y'

**Screenshots**
![image](https://github.com/user-attachments/assets/098eda05-d077-476c-a0b8-86cb82950b3d)

**Expected behavior**
Impersonate to user 'Y' as the same way when the configuration hasn't SSO enabled

**WordPress Environment**
- Website URL: local
- PHP Version: 
- WordPress Version: 6.6.1
- Plugin Version: 3.10.0
- Identity Provider: my own Idp
- Relevant Plugin Settings:

**Comment by timnolte** [comment]
This is not really a bug. This plugin doesn't support use with user switching/impersonation. It is essentially against using SSO/an IDP to impersonate another user.

**Comment by alextochetto** [comment]
> This is not really a bug. This plugin doesn't support use with user switching/impersonation. It is essentially against using SSO/an IDP to impersonate another user.

I agree with you about impersonating another user, but my client needs this feature.
Do you have any tips to jump on this request to help me, this plugin is the only one I found to solve my problem, we are moving from Auth0 IdP to our own IdP.
Could you help me?

Thanks for your time and quick answer
Regards 
Alex

**Comment by JUVOJustin** [comment]
@alextochetto i just tried to reproduce since we do have the same requirement. I cannot reproduce this behavior.

**Comment by alextochetto** [comment]
> @alextochetto i just tried to reproduce since we do have the same requirement. Sadly i can't. Do you have some more details?

@JUVOJustin , how did you configure the environment?
Did you enable the Login Type to Auto Login - SSO in plugin configuration?

1. You need to login with a user using IdP
2. This user must have Administrator profile
3. Then use User Switching plugin to switch user like the image below

![image](https://github.com/user-attachments/assets/daad8bcb-266d-4178-9f1e-d2e411465c71)

The wordpress will try to switch, but keep the actual user logged and cancel the switch
If you need more information let me know

**Comment by alextochetto** [comment]
@timnolte , are you interested in being paid for this enhancement?

**Comment by JUVOJustin** [comment]
@alextochetto you are right i did not configure Auto Login. Will check and give it a try.
